### PR TITLE
Update to new HTTP Proxy API.

### PR DESCRIPTION
**Pull request by trestletech**
@aronatkins please verify: are we OK with this req.pause() call? It seems to be good as of node 10.21, but I'm unclear about the internals of whether or not we might miss any data on the req stream.

(merges into my other PR, not master)

### CODE REVIEW

**Comment by trestletech** [comment]
My concern about dropping data coming from https://github.com/nodejitsu/node-http-proxy/issues/802

**Comment by aronatkins** [comment]
Calling `req.pause` causes the stream to flip out of flowing mode. That looks correct, as you want to do something before continuing to process bytes. 

I wonder if you need `req.resume` in any of the other paths through this code.

File: lib/proxy/http.js [hunk 1]
```diff
@@ -177,7 +178,8 @@ function ShinyProxy(router, schedulerRegistry) {
         req.headers['shiny-shared-secret'] = appWorkerHandle.endpoint.getSharedSecret();
         // Remember that we will null out req and res in the proxy, so be careful
         // trying to subscribe to events on them after we've started this proxy.
-        appWorkerHandle.proxy.proxyRequest(req, res, buffer);
+        appWorkerHandle.proxy.web(req, res);
+        req.resume();
       })
       .fail(function(err) {
         if (acquired){
```
Comment (trestletech) [inline lib/proxy/http.js]: add req.resume here

Comment (trestletech) [inline lib/proxy/http.js]: And line 97, as the internal router would have handled the request and we should unblock the proxy.

**Pull request closed without merging**